src/components/Staff/NavBar.vue

```diff
@@ -19,14 +19,19 @@
 
 <script>
 export default {
-  data: () => ({
-  }),
-  computed: {
-  },
+  data: () => ({}),
+  computed: {},
   methods: {
     async logout() {
-      await this.$store.dispatch('logout');
-      this.$router.push('/');
+      await this.$store.dispatch("logout");
+      this.$router.push("/");
+    },
+    async getRid() {
+      const payload = {
```

Comment:
This function is to load rid in store. hence payload should not be defined here since it always gets rid.
You can put payload in store then u dont even need this method, at created method, call this.$store.dispatch("staff/fetchRid") without payload.